### ISSUE TITLE
`FilePath` rule option to ignore module?

### ISSUE THREAD

**Issue by mockdeep** [backus/rubocop-rspec#519]
I sometimes like to use a top-level application module to namespace internal stuff for clarity, but I don't really want to have to nest my specs under that directory. It would be nice if there was some option to allow ignoring certain modules when resolving the path. For example:

```ruby
# class
module MyApp
  module Matchers
    class HaveTask
      # code here
    end
  end
end

# spec
describe MyApp::Matchers::HaveTask do
  # tests here
end
```

For the above class `HaveTask`, I would want to have the spec file in `spec/matchers/have_task_spec.rb`, but the `FilePath` rule requires that it goes under `spec/my_app/matchers/have_task_spec.rb`.

**Comment by bquorning** [comment]
I think that’s a good idea.

Just to be clear, the suggestion is to be able to define a namespace to ignore, e.g. `MyApp` or `MyApp::Matchers`? It should not be possible to ignore any random module, e.g. `Matchers` and have the file path `spec/my_app/have_task_spec.rb` enforced – right?

**Comment by mockdeep** [comment]
@bquorning for the most part I think that would work for me given the above use case, though another case that we run into occasionally is when we're monkey patching a library that is deeply nested. For example, we have a file `config/initializers/translation_helper.rb` that monkey patches `ActionView::Helpers::TranslationHelper`. For our tests we need to nest it at `spec/initializers/action_view/helpers/translation_helper/t_spec.rb`, which seems a little weird, so it might be nice to be able to ignore the `ActionView::Helpers` part. I'm a little less concerned about that case, though, since I don't have to deal with it all that often, so much as the first case of namespacing my own code.

**Comment by bquorning** [comment]
In my largest codebase, I have `RSpec/FilePath` disabled for 2 specs – and in both of them I test 3rd party code behavior (e.g. `ActionView::Helpers::FormTagHelper`), just like the example you mentioned.

I’m not entirely sure what do to about that case. Adding configuration for what’s often a one-off spec doesn’t seem like a better solution than to just add the path to `Exclude` for `RSpec/FilePath`.

**Comment by mockdeep** [comment]
@bquorning, yeah, I was thinking pretty much the same thing. I think even if I had the option, I'd probably be hesitant to add it to my `.rubocop.yml` file as opposed to either ignoring the file entirely or just deeply nesting it. Sucks to not have it covered, but I can't think of a graceful alternative. Maybe it's good that it's ugly so I'll be motivated to find a way to remove it.
